torchbenchmark/score/compute_score_v2.py

```diff
@@ -0,0 +1,210 @@
+    Model = getattr(module, 'Model')
+    return Model.task
+
+def _parse_test_name(name):
+    """
+    Helper function which extracts test type (eval or train), model,
+    device, and mode from the test full name.
+    """
+    test, model_name, device, mode = re.match(r"test_(.*)\[(.*)\-(.*)\-(.*)\]", name).groups()
+    return (test, model_name, device, mode)
+
+class TorchBenchV2Test:
+    def __init__(self, test_name, test_item):
+        self._name = test_name
+        self._test_type, self._model, self._device, self._mode = _parse_test_name(self._name)
+        self._task = _get_model_task(self._model)
+        self._stable = test_item["stable"]
+        self._norm = test_item["norm"]
+    @property
+    def name(self) -> str:
+        return self._name
+    @property
+    def test_type(self) -> str:
+        return self._test_type
+    @property
+    def model(self) -> str:
+        return self._model
+    @property
+    def device(self) -> str:
+        return self._device
+    @property
+    def mode(self) -> str:
+        return self._mode
+    @property
+    def category(self) -> str:
+        return type(self._task).__name__
+    @property
+    def domain(self) -> str:
+        return self._task.name
+    @property
+    def norm(self) -> float:
+        return self._norm
+    @property
+    def stable(self) -> bool:
+        return self._stable
+
+class TorchBenchV2Suite:
+    def __init__(self, norm):
+        self._tests = []
+        self._tests_dict = {}
+        self._threshold = norm["stable_threshold"]
+        self._target = norm["target"]
+        for test in norm["tests"]:
+            test_item = TorchBenchV2Test(test, norm["tests"][test])
+            self._add_test(test_item)
+    @property
+    def target(self) -> float:
+        return self._target
+    @property
+    def all_stable_tests(self) -> List[TorchBenchV2Test]:
+        return list(filter(lambda x: x.stable, self._tests))
+    @property
+    def threshold(self) -> float:
+        return self._threshold
+    def _add_test(self, test: TorchBenchV2Test):
+        self._tests.append(test)
+        self._tests_dict[test.name] = test
+    def get_test_by_name(self, name) -> TorchBenchV2Test:
+        return self._tests_dict[name]
+
+class TorchBenchScoreV2:
+    # ref_data: the object read from reference YAML file or benchmark json file
+    def __init__(self, ref_data, _spec_file, _target):
+        if not ref_data:
+            with open(TORCHBENCH_V2_REF_DATA) as ref_file:
+                ref_data = yaml.full_load(ref_file)
+        # Build the suite
+        self.suite = TorchBenchV2Suite(self._setup_benchmark_norms(ref_data))
+
+    def _get_test_delta_weight(self, ref_norm, data_norm):
+        delta = (ref_norm - data_norm) / ref_norm
+        # Not a valid signal because it is below threshold
+        if abs(delta) <= self.suite.threshold:
+            return 0.0
+        return delta * 100
```

Comment:
Return the percentage number of delta. For example, if the test regresses `10%`, return `10`.